neo/SmartContract/InteropService.cs

```diff
@@ -180,41 +156,34 @@ protected bool Runtime_CheckWitness(ApplicationEngine engine)
             return true;
         }
 
-        internal void SendNotification(ApplicationEngine engine, UInt160 script_hash, StackItem state)
-        {
-            NotifyEventArgs notification = new NotifyEventArgs(engine.ScriptContainer, script_hash, state);
-            Notify?.Invoke(this, notification);
-            notifications.Add(notification);
-        }
-
-        protected bool Runtime_Notify(ApplicationEngine engine)
+        private static bool Runtime_Notify(ApplicationEngine engine)
         {
-            SendNotification(engine, new UInt160(engine.CurrentContext.ScriptHash), engine.CurrentContext.EvaluationStack.Pop());
+            engine.SendNotification(engine.CurrentScriptHash, engine.CurrentContext.EvaluationStack.Pop());
             return true;
         }
 
-        protected bool Runtime_Log(ApplicationEngine engine)
+        private static bool Runtime_Log(ApplicationEngine engine)
         {
             string message = Encoding.UTF8.GetString(engine.CurrentContext.EvaluationStack.Pop().GetByteArray());
-            Log?.Invoke(this, new LogEventArgs(engine.ScriptContainer, new UInt160(engine.CurrentContext.ScriptHash), message));
+            Log?.Invoke(null, new LogEventArgs(engine.ScriptContainer, engine.CurrentScriptHash, message));
```

Comment:
```suggestion
            Log?.Invoke(engine, new LogEventArgs(engine.ScriptContainer, engine.CurrentScriptHash, message));
```

Reply:
We can pass `engine` if the event is sent from `engine`. For `static` events, just pass `null` as the `sender`.

Reply:
Is static, but if you want to have access to the engine you can't